docs/sorting-algorithms/heap-sort.mdx

```diff
@@ -0,0 +1,326 @@
+            buildHeap(arr, size, i);
+        }
+        long stopTime = System.nanoTime();
+        System.out.println((stopTime - startTime)*0.000000001);
+
+        System.out.println("Time to make sort using max heaps and putting largest element in the end:");
+        startTime = System.nanoTime();
+        // exchanging and extracting elements one by one from the final heap to sort the array
+        for(int i = size-1; i>0; i--) {
+
+            // swap the current root and last term
+            swap(arr, i, 0);
+
+            // doing max heap again
+            buildHeap(arr, i, 0);
+        }
+        stopTime = System.nanoTime();
+        System.out.println((stopTime - startTime)*0.000000001);
+    }
+
+    private static void buildHeap(int arr[], int size, int i) {
+
+        int max = i;
+        int leftNode = 2 * i + 1;
+        int rightNode = 2 * i + 2;
+
+        // for max heap root must be greater than both nodes, so we compare to check this
+        if(leftNode<size && arr[leftNode]>arr[max]) {
+            // left node larger than current root
+            max = leftNode;
+        }
+
+        if(rightNode<size && arr[rightNode]>arr[max]) {
+            // right node larger than left node and current root
+            max = rightNode;
+        }
+
+        if (max != i) {
+            // root is not greatest
+            swap(arr, max, i);
+
+            // since this sub-heap got rearranged we need to rearrange all sub-heaps before this recursively
+            buildHeap(arr, size, max);
+        }
+    }
+
+    private static void swap(int arr[], int pos1, int pos2) {
+        int temp = arr[pos1];
+        arr[pos1] = arr[pos2];
+        arr[pos2] = temp;
+    }
+
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+
+        // taking input array
+        System.out.println("Enter size of array:");
+        int size = sc.nextInt();
+        int arr[] = new int[size];
+        System.out.println("Enter array elements:");
+        for (int i = 0; i < size; i++) {
+            arr[i] = sc.nextInt();
+        }
+
+        // before sorting
+        System.out.println("Array before heap sort:");
+        for (int i = 0; i < arr.length; i++) {
+            System.out.print(arr[i] + " ");
+        }
+        System.out.println();
+
+        long startTime = System.nanoTime();
+        heapSort(arr);
+        long stopTime = System.nanoTime();
+        System.out.println("Time to sort the array:");
+        System.out.println((stopTime - startTime)*0.000000001);
+
+        // after sorting
+        System.out.println("Array after heap sort:");
+        for (int i=0; i<arr.length; i++) {
+            System.out.print(arr[i] + " ");
+        }
+        System.out.println();
+        sc.close();
+    }
+}
+```
+</TabItem>
+
+</Tabs>
+
+# Sample Input and Output
+
+### Input :
+
+```
+Enter elements to be sorted : 5
+30
+50
+10
+20
+40
+```
+### Output :
+```
+[10, 20, 30, 40, 50]
+```
```

Comment:
Use consistent console statements across all the languages

Reply:
done